dotnet/src/Connectors/Connectors.AzureAIInference.UnitTests/Services/AzureAIInferenceChatCompletionServiceTests.cs

```diff
@@ -0,0 +1,328 @@
+        ChatCompletionsClient client = new(this._endpoint, new AzureKeyCredential("api-key"));
+
+        // Act & Assert
+        // Endpoint constructor
+        new AzureAIInferenceChatCompletionService(endpoint: this._endpoint); // Only the endpoint
+        new AzureAIInferenceChatCompletionService(httpClient: httpClient); // Only the HttpClient with a BaseClass defined
+        new AzureAIInferenceChatCompletionService(modelId: "model", endpoint: this._endpoint); // ModelId and endpoint
+        new AzureAIInferenceChatCompletionService(modelId: "model", apiKey: "api-key", endpoint: this._endpoint); // ModelId, apiKey, and endpoint
+        new AzureAIInferenceChatCompletionService(endpoint: this._endpoint, loggerFactory: loggerFactoryMock.Object); // Endpoint and loggerFactory
+
+        // Breaking Glass constructor
+        new AzureAIInferenceChatCompletionService(chatClient: client); // Client
+        new AzureAIInferenceChatCompletionService(chatClient: client, modelId: "model"); // Client
+        new AzureAIInferenceChatCompletionService(chatClient: client, loggerFactory: loggerFactoryMock.Object); // Client
+    }
+
+    [Theory]
+    [InlineData("http://localhost:1234/chat/completions")] // Uses full path when provided
+    [InlineData("http://localhost:1234/v2/chat/completions")] // Uses full path when provided
+    [InlineData("http://localhost:1234")]
+    [InlineData("http://localhost:8080")]
+    [InlineData("https://something:8080")] // Accepts TLS Secured endpoints
+    [InlineData("http://localhost:1234/v2")]
+    [InlineData("http://localhost:8080/v2")]
+    public async Task ItUsesCustomEndpointsWhenProvidedDirectlyAsync(string endpoint)
+    {
+        // Arrange
+        var chatCompletion = new AzureAIInferenceChatCompletionService(modelId: "any", apiKey: null, httpClient: this._httpClient, endpoint: new Uri(endpoint));
+        this._messageHandlerStub.ResponseToReturn = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+        { Content = this.CreateDefaultStringContent() };
+
+        // Act
+        await chatCompletion.GetChatMessageContentsAsync(this._chatHistoryForTest, this._executionSettings);
+
+        // Assert
+        Assert.StartsWith($"{endpoint}/chat/completions", this._messageHandlerStub.RequestUri!.ToString());
+    }
+
+    [Theory]
+    [InlineData("http://localhost:1234/chat/completions")] // Uses full path when provided
+    [InlineData("http://localhost:1234/v2/chat/completions")] // Uses full path when provided
+    [InlineData("http://localhost:1234")]
+    [InlineData("http://localhost:8080")]
+    [InlineData("https://something:8080")] // Accepts TLS Secured endpoints
+    [InlineData("http://localhost:1234/v2")]
+    [InlineData("http://localhost:8080/v2")]
+    public async Task ItPrioritizesCustomEndpointOverHttpClientBaseAddressAsync(string endpoint)
+    {
+        // Arrange
+        this._httpClient.BaseAddress = new Uri(endpoint);
```

Comment:
How do you know that it is prioritizing the custom endpoint over the htttpclient base address, if both are set to Uri(endpoint)?